### PR TITLE
feat: add baseline migration

### PR DESCRIPTION
**Pull request by swapnil-turing**

### CODE REVIEW

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""Baseline schema migration."""
+
+from alembic import op
+
+from app import model
+from app.core.database import Base
+
+
+revision = "0001"
+down_revision = "0001"
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:Marked useful because this comment caught one migration graph mistake that can easily slip during schema bootstrap work. The self-referential `down_revision` would break Alembic ordering and block upgrade resolution for the very first revision.

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    bind = op.get_bind
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:Chose useful here because it pointed to a real runtime failure in the migration path, not just a style issue. Passing the `op.get_bind` function object into `create_all` would make the baseline upgrade crash when Alembic executes it.

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    Base.metadata.create_all(bind=bind, tables=Base.metdata.tables.values())
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:This deserved useful because it found a direct typo in the only code path that creates the schema. The `Base.metdata` access would raise `AttributeError` immediately and stop the migration before any tables are created.

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def downgrade() -> None:
+    Base.metadata.dropall(bind=op.get_bind())
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:Marked useful because it identified one broken downgrade API call in a place we usually exercise less often. Using `dropall` on SQLAlchemy metadata would fail when rolling back this revision and leave the migration non-reversible.

**Comment by swapnil-turing** [comment]
augment review

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
+"""Baseline schema migration."""
+
+from alembic import op
+
+from app import model
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:Marked useful because this comment caught a migration import path mistake that is easy to miss in a small schema bootstrap file. It identified that Alembic would fail before running upgrade() or downgrade() because `app.model` does not exist, so the baseline migration could not be applied at all.

**Comment by swapnil-turing** [comment]
augment review

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+
+revision = "0001"
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    bind = op.get_bind()
+    Base.metadata.create_all(bind=bind, tables=Base.metadata.tables.values())
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:Marked useful because this comment caught a migration design mistake that could quietly change revision `0001` later during a fresh bootstrap. It identified that `Base.metadata.create_all()` would follow future ORM edits, so the baseline schema and downgrade path could drift from the reviewed revision.

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+"""Baseline schema migration."""
+
+from alembic import op
+
+from app import models  # noqa: F401
+from app.core.database import Base
+
+
+revision = "0001"
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py:7]: value:useful; category:bug; feedback:Chose useful because this comment showed the new revision was not runnable in the repository without the expected Alembic wiring around it. It identified that `alembic upgrade head` could fail to locate or execute revision `0001`, which makes the baseline migration unusable for fresh environments.

**Comment by swapnil-turing** [comment]
augment review

File: qualia/backend/alembic.ini [hunk 8]
```diff
@@ -0,0 +1,35 @@
+[alembic]
+script_location = alembic
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic.ini]: value:useful; category:bug; feedback:Marked useful because this comment caught a real path resolution problem in the new Alembic setup before someone tried to run it from the repo root. It identified that the migration config depended on the caller's working directory, which could make basic upgrade commands fail to find the scripts or import path.

File: qualia/backend/alembic/env.py [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+    with context.begin_transaction():
+        context.run_migrations()
+
+
+def run_migrations_online() -> None:
+    configuration = config.get_section(config.config_ini_section, {})
+    database_url = _database_url_value()
+    configuration["sqlalchemy.url"] = database_url
+    if _uses_async_driver(database_url):
+        asyncio.run(run_async_migrations(configuration))
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/env.py]: value:useful; category:bug; feedback:Chose useful because this comment exposed one runtime edge case in the async migration path that is easy to miss during normal CLI testing. It pointed out that programmatic migrations inside an active event loop would crash with RuntimeError, so async test and service flows could not reuse this migration entrypoint safely.

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+        sa.ForeignKeyConstraint(["question_id"], ["questions.id"], ondelete="CASCADE"),
+        sa.ForeignKeyConstraint(["submission_id"], ["submissions.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint("submission_id", "question_id", name="uq_submission_answer_submission_question"),
+    )
+    op.create_index(op.f("ix_submission_answers_question_id"), "submission_answers", ["question_id"], unique=False)
+    op.create_index(op.f("ix_submission_answers_submission_id"), "submission_answers", ["submission_id"], unique=False)
+
+
+def downgrade() -> None:
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py:200]: value:useful; category:bug; feedback:Selected useful because this comment found one cleanup gap in the baseline downgrade that would leave the database in a partially migrated state. It identified that PostgreSQL enum types would remain after downgrade and could break a later upgrade when the migration tries to recreate the same named types.

**Comment by swapnil-turing** [comment]
augment review

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 11]
```diff
@@ -0,0 +1,240 @@
...
+        sa.Column("reviewer_id", sa.Uuid(), nullable=False),
+        sa.Column("created_at", sa.DateTime(timezone=True), server_default=sa.func.now(), nullable=False),
+        sa.Column("status", submission_status_enum, server_default=sa.text("'draft'"), nullable=False),
+        sa.Column("started_at", sa.DateTime(timezone=True), server_default=sa.func.now(), nullable=False),
+        sa.Column("submitted_at", sa.DateTime(timezone=True), nullable=True),
+        sa.Column("last_saved_at", sa.DateTime(timezone=True), server_default=sa.func.now(), nullable=False),
+        sa.ForeignKeyConstraint(["form_cycle_id"], ["form_cycles.id"]),
+        sa.ForeignKeyConstraint(["reviewer_id"], ["users.id"]),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint("form_cycle_id", name="uq_submissions_form_cycle_id"),
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py:142]: value:incorrect-but-reasonable; category:bug; feedback:This was reasonable to raise because the `submissions` table carries `reviewer_id`, so the row shape can look like one submission per reviewer on first reading. The comment assumed multi-reviewer submissions, but this backend intentionally permits only one reviewer assignment and one submission per form cycle, so the current constraint matches the supported behavior.

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 12]
```diff
@@ -0,0 +1,240 @@
...
+    op.drop_index(op.f("ix_form_cycles_status"), table_name="form_cycles")
+    op.drop_index(op.f("ix_form_cycles_is_published"), table_name="form_cycles")
+    op.drop_index(op.f("ix_form_cycles_created_by_id"), table_name="form_cycles")
+    op.drop_table("form_cycles")
+
+    op.drop_index(op.f("ix_users_role"), table_name="users")
+    op.drop_index(op.f("ix_users_is_active"), table_name="users")
+    op.drop_table("users")
+
+    if op.get_bind().dialect.name == "postgresql":
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:Marked useful because this caught one migration-only failure that is easy to miss when upgrades and downgrades are tested only against a live database. The unconditional `op.get_bind()` call can break `alembic downgrade --sql` in offline mode, so the baseline downgrade would fail to emit SQL for PostgreSQL environments.

**Comment by swapnil-turing** [comment]
augment review

**Comment by swapnil-turing** [comment]
augment review

File: qualia/backend/alembic/env.py [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+from logging.config import fileConfig
+import threading
+
+from alembic import context
+from sqlalchemy import engine_from_config, pool
+from sqlalchemy.ext.asyncio import async_engine_from_config
+
+from app import models  # noqa: F401
+from app.core.config import _database_url
+from app.core.database import Base
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/env.py]: value:useful; category:bug; feedback:Marked useful because this comment caught one migration setup problem that was easy to miss in this PR. It identified that Alembic metadata loading was creating the async engine on import, which could break migrations when the configured URL uses the sync path.

File: qualia/backend/alembic/versions/0001_baseline_schema.py [hunk 14]
```diff
@@ -0,0 +1,243 @@
...
+    op.drop_index(op.f("ix_form_cycles_status"), table_name="form_cycles")
+    op.drop_index(op.f("ix_form_cycles_is_published"), table_name="form_cycles")
+    op.drop_index(op.f("ix_form_cycles_created_by_id"), table_name="form_cycles")
+    op.drop_table("form_cycles")
+
+    op.drop_index(op.f("ix_users_role"), table_name="users")
+    op.drop_index(op.f("ix_users_is_active"), table_name="users")
+    op.drop_table("users")
+
+    if context.is_offline_mode():
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply qualia/backend/alembic/versions/0001_baseline_schema.py]: value:useful; category:bug; feedback:Chose useful because this comment found one real downgrade gap in the offline PostgreSQL path. It pointed out that `alembic downgrade --sql` would skip enum type drops, leaving leftover types that can break a later upgrade or keep the schema inconsistent.

**Comment by swapnil-turing** [comment]
augment review
